docs/source/manual/testing.rst

```diff
@@ -365,24 +365,27 @@ By creating a DropwizardTestSupport instance in your test you can manually start
     public class LoginAcceptanceTest {
 
         public static final DropwizardTestSupport<TestConfiguration> SUPPORT =
-                new DropwizardTestSupport<TestConfiguration>(MyApp.class, ResourceHelpers.resourceFilePath("my-app-config.yaml"));
+                new DropwizardTestSupport<TestConfiguration>(MyApp.class,
+                    ResourceHelpers.resourceFilePath("my-app-config.yaml"),
+                    ConfigOverride.config("server.applicationConnectors[0].port", "0") // create random port
+                );
 
         @BeforeClass
         public void beforeClass() {
-          SUPPORT.before();
+            SUPPORT.before();
         }
 
         @AfterClass
         public void afterClass() {
-          SUPPORT.after();
+            SUPPORT.after();
         }
 
         @Test
         public void loginHandlerRedirectsAfterPost() {
-            Client client = new JerseyClientBuilder(SUPPORT.getEnvironment()).build("test client");
+            Client client = new JerseyClientBuilder().build();
```

Comment:
Are you using the dropwizard provided JerseyClientBuilder? That should support environment.